librz/include/rz_analysis.h

```diff
@@ -623,6 +623,7 @@ typedef struct rz_analysis_t {
 	RzList *leaddrs;
 	RzArchTarget *arch_target;
 	RzArchPlatformTarget *platform_target;
+	HtPP *ht_global_var; // global variables
```

Comment:
In addition to the name-keyed hashtable, we will need a (non-augmented) RBtree of vars keyed by their address since variables have a certain size. Then querying for a variable that contains a certain address becomes possible, which will be quite important in some cases.

Reply:
Yes, this is a great point

Reply:
Please note, that size might change depending on the type set, thus during the "set new type" operation or type propagation we would need to update the RBTree as well.

Reply:
Actually we don't have to update it. The RBTree (when not augmented, which is fine here) just saves the origin address. Global variables must not overlap. But even if they do we can handle it gracefully and emit warnings.
Then to get the variable that contains a given address, we find the highest variable whose address is less than or equal to the given one. This is the only one that could potentially contain this address so all we have to do is do one interval check against it and we are done.

Reply:
Done, please take a look. @thestr4ng3r  @XVilka